src/services/data-provider-link-parser.test.ts

```diff
@@ -73,6 +74,25 @@ describe('data-provider-link-parser', () => {
     expect(result).toEqual(expectedResult);
   });
 
+  it('should match url on both path and name', async () => {
+    const expectedProjectData = generateGitlabProject({
+      id: 1,
+      name: projectName1,
+      web_url: 'https://gitlab.com/test/repo-name',
+    });
+    mockedGetGroupIds.mockResolvedValue([1]);
+    storage.getSecret.mockResolvedValueOnce(testToken1);
+
+    mockedGetOwnedProjectsBySearchCriteria.mockResolvedValueOnce([
+      generateGitlabProject({ id: 2, name: projectName1Deleted, web_url: 'https://gitlab.com/test/repo-name-deleted' }),
+      expectedProjectData,
+    ]);
+
+    const result = await getProjectDataFromUrl(mockProjectUrl);
+    const expectedResult = { project: expectedProjectData, groupToken: testToken1 };
+    expect(result).toEqual(expectedResult);
+  });
+
```

Comment:
<img width="592" alt="Screenshot 2023-05-12 at 4 02 33 PM" src="https://github.com/atlassian-labs/gitlab-for-compass/assets/120515506/e99e58d6-8e0a-49e6-bf1c-333c7689b353">
This test fails without the strict path and name matching because Gitlab projects search returns `https://gitlab.com/test/repo-name-deleted` as an option when the project name query is "repo-name", and the webUrl includes the path of `/test/repo-name`